### ISSUE TITLE
Slow down rate of incoming messages

### ISSUE THREAD

**Issue by uqs** [ORi0N/comfoconnect_openhab_gw#1]
Hey there, I successfully got this to work on my ComfoAir Q350 and it's publishing fine to MQTT (and OpenHAB) and I can see the graphs in Grafana. So far so good, however the rate of information is rather high and it's flooding all logs and the timeseries database with an unneeded granularity/frequency.

There's a "sleep(5)" in your code, but changing that doesn't slow the rate down at all (which is much higher anyway).

I'm not quite sure what is driving this high rate and how to slow it down, my assumption would be that the bridge is invoking the callback too often, but what determines the rate at which the bridge is seeing messages?

**Comment by ORi0N** [comment]
Yes, it has nothing to do with the sleep(5) which is actually pointless in a way. I only used it for polling the unit's status (manual/auto & level) and report if it changed. I don't use it anymore now in my installation.

The flowrate is indeed induced from the callback method. And the bridge is driving the callback. You could slow it down by analyzing the message and only report to mqtt if it changed by a certain factor of lets say 5%/10%. Then you won't get all the subtle changes in fan speed for example.

I simply allow all messages to be pushed to MQTT anyway. It's a message queuing bus, so it can handle a high flow of messages. I limited the persistence from OpenHAB to InfluxDB to 'every minute' instead of 'every update/change' so that I get only 1 value per minute in my graphs.

**Comment by uqs** [comment]
That's what I figured, I guess I'll gate the MQTT updates to 1 change per 15s or so. The thing is I'm running this on a Raspberry Pi and some other OpenHAB events/triggers are getting delayed, not sure the high rate of updates via mosquitto is to blame.

In any case, it should slow down the "log spam" and stop useless work from happening.

If you're no longer using it, is the code on github actually the latest? Would you mind uploading another snapshot maybe?

Also, do you take pull requests?

**Comment by HalliHalloSchatz** [comment]
I am interested. I will slow down the update rate.
I am not a programmer, but I will try. 
When it is working, I try to bring back my code.

**Comment by ORi0N** [comment]
If you're also using Node-Red in your installation you could actually use the 'polling' method to monitor the spammed topics and then post them every x seconds to the topic you wish to monitor with OpenHAB and/or report on in Grafana. This allows you to set the internal also very specifically per message topic :)

Anyway, I think there are better extensions than the one I created several years ago. Just google for them them... There are several topics on them in the community. 
Or just use Home Assistant. For example I'm using Home Assistant AND openHAB side by side because they both have their strengths.

**Comment by HalliHalloSchatz** [comment]
I have no idea what Node-Red is.
I use edomi and mqtt.

Can you send me a few links, than I will try to install them.
